### ISSUE TITLE
TODO: implement lazy load

### ISSUE THREAD

**Issue by netrules** [iconify/iconify#38]
Hello,

I have used your framework for the first time today, and in comparison to other similar frameworks (devicon & font-mfizz), I have found it to be the most performant capable. Your library/framework seems to be the monst maintained and lightweight, not to mention complete with all my UI needs. However, there is something neither of those networks support, and yours doesn't either, which I would like to suggest: **lazy loading for setting svg content**.

It can be used in the following ways:
- on events (injectable prototype events or dom events)
- user interactivity (native web performance lazy loading [which doesn't handle svg yet](https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading))
- trigger methods from configuration (personal user code design)

I would like to hear back from you, and know if there is any way to approach the subject currently. You can see an example of how I'm using your amazing framework here: https://sgonzalez.tech/about/ ... By the time of writing this review, I'm still using the other two libraries due to time constraint, but will probably phase completely to using this only.

P.S.: I found it extremely nice how when my script is disabled (and thus, svg paths not loaded yet), the data-icon text is still showing. Maybe this is a sign of readers being capable to understand what the icons represent? Anyway, that's cool.

Thanks!
### tl;dr: _lazy loading for svg pending_

**Comment by netrules** [comment]
Also, is there a way to make a local svg cache for when I want to enable a scriptless way to present the data, like [polyfills](https://polyfill.io/v3/url-builder), maybe?

**Comment by cyberalien** [comment]
Thanks! Glad to hear you like it!

By lazy loading, do you mean injecting node after icon data is available? It is already possible with function `Iconify.loadIcons()`. Documentation for that function with several examples, including async icon loading: https://docs.iconify.design/implementations/svg-framework/load-icons.html

That documentation is for version 2 of SVG framework, so you'll need to update to version 2.0. See https://docs.iconify.design/implementations/svg-framework/

As for cache, it is also available. You can add icon bundles using function `Iconify.addCollection()`: https://docs.iconify.design/implementations/svg-framework/add-collection.html

Documentation that explains how bundles work and several examples of bundle build scripts: https://docs.iconify.design/sources/bundles/

**Comment by netrules** [comment]
Way ahead of me! _I should've read the damn docs_.  I'm sorry, I didn't realize how neatly documented everything is for this framework. I'm really impressed, thank you.
